### ISSUE TITLE
Layout/MultilineOperationIndentation is missing some cases

### ISSUE THREAD

**Issue by pirj** [rubocop/rubocop#7592]
Layout/MultilineOperationIndentation is missing cases when `\` is used as string concatenation.

```ruby
# 1.rb
puts 'a' +
'b'

puts 'a' \
'b'

puts "#{1}" \
'b'

puts 'a' \
"#{2}"

puts "#{1}" \
"#{2}"
```

--------

## Expected behavior

All the offences are detected.

## Actual behavior

Only the first statement is detected as offensive.

```
1.rb:2:1: C: Layout/MultilineOperationIndentation: Align the operands of an expression spanning multiple lines.
'b'
^^^
```

## Steps to reproduce the problem

```sh
rubocop --only Layout/MultilineOperationIndentation 1.rb
```

## RuboCop version

```
$ [bundle exec] rubocop -V
0.78.0 (using Parser 2.6.5.0, running on ruby 2.5.5 x86_64-darwin18)
```

**Comment by marcandre** [comment]
Is this really a bug though? `/` is not an operator, it just escape the newline. I'd say this falls outside of the cop's jurisdiction:

> This cop checks the indentation of the right hand side operand in binary operations that span more than one line.

**Comment by pirj** [comment]
It might not be its business to take care of `\`, but the indentation looks broken. This may be a more generic issue than just this cop, might equally affect several cops.

To be honest, I agree with you that `\` still makes the same **line**, but it's on different **rows**.
To my understanding, the indentation should consider rows, not lines (I hope I'm not mixing those two terms up), e.g.
```ruby
class A
  def m
    puts "a" \
"b"
  end
end
```

from my point of view should raise at least some offence. I might be just me, but aestetically this code is broken.

**Comment by marcandre** [comment]
Oh, I agree it's not a great alignment. I just meant to ask: is it this cop's responsibility? I'm still pretty ignorant of all the different cops we have 😀

**Comment by pirj** [comment]
Got it.
Answering this question directly: yes, I believe it's this cop's responsibility to raise an offence for those code examples.
But since it might be a common problem in indentation cops not being aware of `\`.

Still, I think it's a very minor issue, I've only seen this problem once in the wild, fixed it by hand, and no cop complained about the correct indentation. No objections if this is closed. It's just my OCD that pushed me to file this ticket.
